### PR TITLE
Cypress Cloud integration

### PR DESCRIPTION
**Pull request by roicarrera**
added integration with cypress cloud. Has to be set manually, explained in the readme. 
Changed agent node version to node 18

### CODE REVIEW

File: e2e-cypress/files/README.md [hunk 1]
```diff
@@ -43,6 +43,34 @@ In order to generate one xml report per test type (installation, integration and
 With Cypress 12 version is now available `cy.origin()` that allows you to handle redirections. This funcionality eases the login handling.
 See `./support/e2e.ts` for a generic login example.
 
+## Cypress Cloud
```
Comment (cschweikert) [inline e2e-cypress/files/README.md:46]: I would move the whole section into the documentation of cypress (https://github.com/opendevstack/ods-quickstarters/blob/master/docs/modules/quickstarters/pages/e2e-cypress.adoc). The focus of the `README.md` should be more on getting started. Maybe only leave a short note about general support for Cypress Cloud with a link to the documentation.

File: e2e-cypress/files/README.md [hunk 2]
```diff
@@ -43,6 +43,34 @@ In order to generate one xml report per test type (installation, integration and
...
+
+To use Cypress Cloud, follow these steps:
+1. **Request access.** In order to access Cypress Cloud, you will need to request access to the AAD group. This can be done through MyServices using the Active Directory Group - Administration Request form. Once access has been granted, you will be able to log into https://cloud.cypress.io/login using your mail.
```
Comment (cschweikert) [inline e2e-cypress/files/README.md:49]: This point refers to internal processes of a specific organization. I would not mention those within an open source project. Maybe rephrase this and the following point to be more general.

File: e2e-cypress/Jenkinsfile.template [hunk 3]
```diff
@@ -4,15 +4,19 @@
...
   dockerRegistry = env.DOCKER_REGISTRY
+  cypressRecordKey = env.CYPRESS_RECORD_KEY
 }
 
 odsComponentPipeline(
   podContainers: [
       containerTemplate(
         name: 'jnlp',
-        image: "${dockerRegistry}/ods/jenkins-agent-nodejs16:@shared_library_ref@",
+        image: "${dockerRegistry}/ods/jenkins-agent-nodejs18:@shared_library_ref@",
```
Comment (cschweikert) [inline e2e-cypress/Jenkinsfile.template:14]: The update to NodeJS 18 should be mentioned in the changelog as well.

**Review by cschweikert — CHANGES_REQUESTED** [review]

File: e2e-cypress/files/README.md (hunk 2 above)
Comment (braisvq1996) [reply e2e-cypress/files/README.md]: This is pending to be changed from their side already.

File: e2e-cypress/Jenkinsfile.template (hunk 3 above)
Comment (braisvq1996) [reply e2e-cypress/Jenkinsfile.template:14]: Seems that @roicarrera already modified the Changelog accordingly.

File: e2e-cypress/files/README.md (hunk 1 above)
Comment (braisvq1996) [reply e2e-cypress/files/README.md:46]: Done ;)

**Review by cschweikert — APPROVED** [review]
